paychmgr/settle_test.go

```diff
@@ -0,0 +1,72 @@
+package paychmgr
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	tutils "github.com/filecoin-project/specs-actors/support/testing"
+	ds "github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPaychSettle(t *testing.T) {
+	ctx := context.Background()
+	store := NewStore(ds_sync.MutexWrap(ds.NewMapDatastore()))
+
+	expch := tutils.NewIDAddr(t, 100)
+	expch2 := tutils.NewIDAddr(t, 101)
+	from := tutils.NewIDAddr(t, 101)
+	to := tutils.NewIDAddr(t, 102)
+
+	sm := newMockStateManager()
+	pchapi := newMockPaychAPI()
+	defer pchapi.close()
+
+	mgr, err := newManager(sm, store, pchapi)
+	require.NoError(t, err)
+
+	amt := big.NewInt(10)
+	_, mcid, err := mgr.GetPaych(ctx, from, to, amt)
+	require.NoError(t, err)
+
+	time.Sleep(10 * time.Millisecond)
```

Comment:
these sleeps make me nervous. If we start seeing random test failures, we're gonna have to do something different

Reply:
+1 on this -- I'd like to see this fixed either way in a future PR -- I prefer no time.Sleep's outside of integration tests. It looks like what you'd want to do is write to channels in the mockPaychAPI methods so you can simply delay until they are called.

Reply:
Resolved by https://github.com/filecoin-project/lotus/pull/2869